sympy/calculus/util.py

```diff
@@ -10,6 +10,131 @@
 from sympy.functions.elementary.miscellaneous import Min, Max
 
 
+def continuous_domain(f, symbol, domain):
+    """
+    Returns the intervals in the given domain for which the function is continuous.
+    This method is limited by the ability to determine the various
+    singularities and discontinuities of the given function.
+
+    Examples
+    ========
+    >>> from sympy import Symbol, S, tan, log, pi, sqrt
+    >>> from sympy.sets import Interval
+    >>> from sympy.calculus.util import continuous_domain
+    >>> x = Symbol('x')
+    >>> continuous_domain(1/x, x, S.Reals)
+    (-oo, 0) U (0, oo)
+    >>> continuous_domain(tan(x), x, Interval(0, pi))
+    [0, pi/2) U (pi/2, pi]
+    >>> continuous_domain(sqrt(x - 2), x, Interval(-5, 5))
+    [2, 5]
+    >>> continuous_domain(log(2*x - 1), x, S.Reals)
+    (1/2, oo)
+
+    """
+    from sympy.solvers.inequalities import solve_univariate_inequality
+    from sympy.solvers.solveset import solveset, _has_rational_power
+
+    if domain.is_subset(S.Reals):
+        constrained_interval = domain
+        for atom in f.atoms(Pow):
+            predicate, denom = _has_rational_power(atom, symbol)
+            constraint = S.EmptySet
+            if predicate and denom == 2:
+                constraint = solve_univariate_inequality(atom.base >= 0,
+                                                         symbol).as_set()
+                constrained_interval = Intersection(constraint,
+                                                    constrained_interval)
+        for atom in f.atoms(log):
+            constraint = solve_univariate_inequality(atom.args[0] > 0,
+                                                     symbol).as_set()
+            constrained_interval = Intersection(constraint,
+                                                constrained_interval)
+        domain = constrained_interval
```

Comment:
Rather in general every logical block should be followed by a blank line, it makes the code easier to read.